### PR TITLE
Update `INotifyCollectionChanged` events in `MenuItems` and `FooterMenuItems` of the `NavigationView`

### PR DESCRIPTION
**Pull request by ioswald**
## Pull request type

Please check the type of change your PR introduces:

- [ ] Update
- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes

## What is the current behavior?

If you modify a databound observable collection to navigationview menu/footer items and then modify that list later in the program, the ui will not update with the new items.

Issue Number: N/A

## What is the new behavior?
Ui now updates when binding source is updated.

- added source binding list change event handling for navigationview menu/footer items.
- Added collection action handling of menu/footer items.
-
-

### CODE REVIEW

**Comment by ioswald** [comment]
Fixed stackoverflow due to recursive call from collection change event when updating from binding source.

**Comment by crippledfaith** [comment]
Hi please look at (https://github.com/lepoco/wpfui/pull/1092) You don't need to add the items to another collection as it is an observable collection, any change to the collection will change the items.

**Comment by ioswald** [comment]
The initial items get added to the observable collection and update when you set the binding source, but you aren't adding a binding to the binding source so if the binding source changes after the initial binding is done; the items do not update.

File: src/Wpf.Ui/Controls/NavigationView/NavigationView.Properties.cs [hunk 1]
```diff
@@ -450,6 +450,45 @@ public Thickness FrameMargin
...
+    {
+        if (ReferenceEquals(sender, collection))
+        {
+            return;
+        }
+
+        switch (e.Action)
+        {
+            case NotifyCollectionChangedAction.Add:
+                foreach (var item in e.NewItems)
```
Comment (pomianowski) [inline src/Wpf.Ui/Controls/NavigationView/NavigationView.Properties.cs:463]: Use explicit type

**Pull request merged**